### PR TITLE
chmode: centralise + test mode limits

### PR DESCRIPTION
**Pull request by edk0**
One tangible effect of this is that mode changes in oper overrides will be more accurate.

### CODE REVIEW

File: tests/chmode1.c [hunk 1]
```diff
@@ -66,6 +66,41 @@ test_chmode_parse(void)
 	remove_hook("get_channel_access", chmode_access_hook);
 }
 
+void
+test_chmode_limits(void)
+{
+	char chmode_buf[2 + MAXMODEPARAMS + 1] = "+";
```
Comment (dwfreed) [inline tests/chmode1.c:72]: Why is this 2 + ... instead of 1 + ... ?

**Review by dwfreed — COMMENTED** [review]
Just one question that may be due to me not being able to brain

File: tests/chmode1.c (hunk 1 above)
Comment (edk0) [reply tests/chmode1.c:72]: The string is `+{'l' * (MAXMODEPARAMS+1)}\0` and I thought `1 + 1` looked silly

Comment (dwfreed) [reply tests/chmode1.c:72]: Aha, missed the for loop after was using `MAXMODEPARAMS + 1`

**Review by dwfreed — APPROVED** [review]

**Pull request merged**